org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolService.java

```diff
@@ -119,6 +119,13 @@ public List<? extends Location> getDefinitions(XtextResource resource, int offse
 		return locations;
 	}
 
+	@Deprecated
+	public List<? extends Location> getDefinitions(XtextResource resource, int offset,
```

Comment:
This is no longer invoked, is it?

Reply:
That is right, it is no longer invoked. Since it is a public method, I set it as deprecated instead of removing it. It can also add //forRemoval or I could just delete it, if you prefer.

Reply:
what i dont understand. why cant the old method simply be changed?

Reply:
I did not change it because it is a public method and I do not know if someone relies on the old behaviour. That is why I leave the method in place with a deprecation, just in case. Maybe I am just to cautious :), if we do not need to be that careful, I would just remove the method or change it so that it does what we need, as you prefer.

Reply:
but if it is not called subclasses might break anyway ?!?

Reply:
I have the feeling we are not thinking about the same kind of breakage.
`DocumentSymbolService.getDefinitions(XtextResource, int, IResourceAccess, CancelIndicator)` is public (not protected), so it is not only about subclasses. One could call that method from somewhere else. I have decided to keep this method as is because of that.

Regarding subclasses, `getReferences` and `getDefinitions(Document, XtextResource, DefinitionParams, IResourceAccess, CancelIndicator)` are the methods called from `LanguageServerImpl`. The first one will behave as before, the second one will newly return links where there are cross-references. That is a change in behaviour but I believe it is the correct behaviour for the request `textDocument/definitions`. Since both methods used before `DocumentSymbolService.getDefinitions(XtextResource, int, IResourceAccess, CancelIndicator)` and that will return too much, I do not know how to avoid some kind of breakage here.

I feel it is best to remove usage of the conflicting method and deprecate it, but we could do something different. What would you if you where to correct `getDefinitions(Document, XtextResource, DefinitionParams, IResourceAccess, CancelIndicator)` to return only definitions where the symbol is actually a cross-refs?

Reply:
i rather completely remove the old stuff
@szarnekow what do you think?

Reply:
The API is question is clearly designed to be called from framework code instead of user-code. Please don't keep methods around as deprecated that are no longer working.

Reply:
Thanks, I have done that and added a test case.